### PR TITLE
[MS-797] Update CommCareIdentityDataSource to return enrollment recods based on template format

### PR DESCRIPTION
**Pull request by meladRaouf**
Modify CommCareIdentityDataSource to retrieve enrollment records according to the template format.

### CODE REVIEW

File: infra/enrolment-records-store/src/test/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSourceTest.kt [hunk 1]
```diff
@@ -146,25 +172,28 @@ class CommCareIdentityDataSourceTest {
...
-                expectedFingerprint.fingerIdentifier == actualFingerprint.fingerIdentifier &&
-                expectedFingerprint.templateQualityScore == actualFingerprint.templateQualityScore &&
-                expectedFingerprint.template.contentEquals(actualFingerprint.template) &&
-                expectedFingerprint.format == actualFingerprint.format
-            } .all { it }
-        } .all { it }
+        assertEquals(1, actualIdentities.size)
+        val areContentsEqual =
+            expectedFingerprintIdentities.filter { identity -> identity.fingerprints.any { it.format == templateFormat } }
+                .zip(actualIdentities) { expected, actual ->
```
Comment (luhmirin-s) [inline infra/enrolment-records-store/src/test/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSourceTest.kt:190]: Minor formatting nitpick: this block is extremely confusing with its wrapping and mixing of comparisons and function chains.

Comment (luhmirin-s) [reply infra/enrolment-records-store/src/test/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSourceTest.kt:190]: Maybe you could make use of https://truth.dev/api/1.1/index.html?com/google/common/truth/Correspondence.html to simplify the checks here?

**Review by luhmirin-s — APPROVED** [review]

**Comment by BurningAXE** [comment]
Should this target `2024.2.1` or `main`?

**Comment by meladRaouf** [comment]
> Should this target `2024.2.1` or `main`?

I believe it's a simple fix, and we should be able to include it in the release.

**Review by BurningAXE — APPROVED** [review]
Please make sure to redo the tests with latest build if we include it in 2024.2.1

**Pull request merged**